### PR TITLE
[4.1] Adapter trait unification

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
This is a followup pr of #35788 which unifies the usage across the media manager to load the adapters.

### Testing Instructions
Open the media manager in the back end and browse around and upload delete images.

### Actual result BEFORE applying this Pull Request
All is working.

### Expected result AFTER applying this Pull Request
All is working.

### CODE REVIEW

File: administrator/components/com_media/src/Provider/ProviderManagerHelperTrait.php [hunk 1]
```diff
@@ -104,11 +89,48 @@ private function getProvider(String $id): ProviderInterface
...
 	{
 		return $this->getProviderManager()->getAdapter($name);
 	}
 
+	/**
+	 * Returns an array with the adapter name as key and the path of the file.
+	 *
+	 * @return  array
+	 *
+	 * @throws  \Exception
```
Comment (PhilETaylor) [inline administrator/components/com_media/src/Provider/ProviderManagerHelperTrait.php]:
```suggestion
	 * @throws  \InvalidArgumentException
```

File: administrator/components/com_media/src/Provider/ProviderManagerHelperTrait.php [hunk 2]
```diff
@@ -104,11 +89,48 @@ private function getProvider(String $id): ProviderInterface
...
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	protected function resolveAdapterAndPath(String $path): array
+	{
+		$result = [];
+		$parts = explode(':', $path, 2);
+
+		// If we have 2 parts, we have both an adapter name and a file path
+		if (\count($parts) == 2)
```
Comment (PhilETaylor) [inline administrator/components/com_media/src/Provider/ProviderManagerHelperTrait.php]:
```suggestion
		if (\count($parts) === 2)
```

**Comment by dgrammatiko** [comment]
I have tested this item :white_check_mark: successfully on bfe7daa5a138994539f3374d6d7224490936e46a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36097">issues.joomla.org/tracker/joomla-cms/36097</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on bfe7daa5a138994539f3374d6d7224490936e46a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36097">issues.joomla.org/tracker/joomla-cms/36097</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36097">issues.joomla.org/tracker/joomla-cms/36097</a>.</sub>

**Comment by bembelimen** [comment]
Thx

**Pull request merged**